infrastructure/tari_script/src/script.rs

```diff
@@ -522,20 +529,25 @@ impl TariScript {
         #[allow(clippy::mutable_key_type)]
         let mut sig_set = HashSet::new();
 
+        let mut agg_pub_key = RistrettoPublicKey::default();
         for s in &signatures {
             for (i, pk) in public_keys.iter().enumerate() {
                 if !sig_set.contains(s) && !key_signed[i] && s.verify_challenge(pk, &message) {
                     key_signed[i] = true;
                     sig_set.insert(s);
+                    agg_pub_key = agg_pub_key + pk;
                     break;
                 }
             }
             if !sig_set.contains(s) {
-                return Ok(false);
+                return Ok(None);
```

Comment:
As an honest question, entering the current scope, means that the signature `s` hasn't signed the challenge set by `pk` and `message`. Shouldn't we error in this case, as this might be a case of a malicious execution ? Moreover, it would also allow to distinguish the case in which not enough signatures have been provided.

Reply:
Error conditions are managed by the caller branch in `fn execute_opcode(`. It depends on the design of the opcode response.